libvmaf/src/thread_pool.c

```diff
@@ -156,7 +156,8 @@ int vmaf_thread_pool_wait(VmafThreadPool *pool)
     if (!pool) return -EINVAL;
 
     pthread_mutex_lock(&(pool->queue.lock));
-    while((!pool->stop && pool->n_working) || (pool->stop && pool->n_threads))
+    while((!pool->stop && (pool->n_working || pool->queue.head)) ||
```

Comment:
Just noting that it is probably logically sufficient to just check pool->queue.head, although I'll wait for someone closer to the code to comment.

(If there are other potential fixes, I can test in an environment where failures are likley)